javascript/diff_match_patch_uncompressed.js

```diff
@@ -68,27 +68,16 @@ var DIFF_EQUAL = 0;
 
 /**
  * Class representing one diff tuple.
- * Attempts to look like a two-element array (which is what this used to be).
+ * Constructor returns an actual two-element array, to allow destructuring
+ * See https://github.com/JackuB/diff-match-patch/issues/14 for details
  * @param {number} op Operation, one of: DIFF_DELETE, DIFF_INSERT, DIFF_EQUAL.
  * @param {string} text Text to be deleted, inserted, or retained.
  * @constructor
  */
 diff_match_patch.Diff = function(op, text) {
-  this[0] = op;
-  this[1] = text;
+  return [op, text];
```

Comment:
I believe this will change the signature of the function in a more severe way than it looks.

before the change it will return `true` for `instanceof diff_match_patch.Diff` but afterwards it won't.

probably best to preserve the existing behavior for now to avoid breaking code relying on the existing interface. it's hard to tell how far this code reaches, but it's fairly universal, so I'm sure most parts of the interface are locked.

the intended behavior can be added with `Symbol.iterator` though I'm not sure what that means for backwards compatibility either. we probably need to establish some guidelines going forward. luckily this has been supported in most browsers since 2016

Reply:
Symbol.iterator was mentioned in the original issue about this, it's been a part of every JavaScript environment except IE11 for probably 7 years. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/iterator#browser_compatibility

Microsoft don't support IE11 anymore, I suspect anyone still deploying for it is already transpiling their code 🤷‍♂️

Reply:
yeah let's leave it in then. if anyone is importing this into `npm` projects they should have something supported within the past eight years.

later on I'd be for converting the `Function` interfaces to `class` syntax. for now though we can keep the `Diff` identity while offering destructuring